static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -35,10 +38,15 @@ function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorP
   const currentUser: User = useUser();
   const dashboardCreator: User | undefined = dashboard.createdBy;
   const isCurrentUserDashboardOwner = dashboardCreator?.id === currentUser.id;
-  const {teams} = useTeamsById();
-  const teamIds: string[] = Object.values(teams).map(team => team.id);
+
+  // Retrieves teams from the team store, which may contain only a subset of all teams
+  const {teams: teamsToRender} = useTeamsById();
+  const {onSearch} = useTeams();
```

Comment:
`useTeams()` is deprecated but I'm using it here for the `onSearch` function. https://github.com/getsentry/sentry/blob/44e82becd67e9c8da69d6e8e5ff2a5f98e5b4790/static/app/utils/useTeams.tsx#L162